includes/api/class-wc-rest-report-sales-controller.php

```diff
@@ -0,0 +1,395 @@
+ *
+ * @package WooCommerce/API
+ * @extends WP_REST_Controller
+ */
+class WC_REST_Report_Sales_Controller extends WP_REST_Controller {
+
+	/**
+	 * Endpoint namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace = 'wc/v1';
+
+	/**
+	 * Route base.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'reports/sales';
+
+	/**
+	 * Report instance.
+	 *
+	 * @var WC_Admin_Report
+	 */
+	protected $report;
+
+	/**
+	 * Register the routes for coupons.
+	 */
+	public function register_routes() {
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Check whether a given request has permission to read report.
+	 *
+	 * @param  WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|boolean
+	 */
+	public function get_items_permissions_check( $request ) {
+		if ( ! wc_rest_check_manager_permissions( 'reports', 'read' ) ) {
+			return new WP_Error( 'woocommerce_rest_cannot_view', __( 'Sorry, you cannot list resources.', 'woocommerce' ), array( 'status' => rest_authorization_required_code() ) );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Get sales reports.
+	 *
+	 * @param WP_REST_Request $request
+	 * @return array|WP_Error
+	 */
+	public function get_items( $request ) {
+		$data   = array();
+		$item   = $this->prepare_item_for_response( null, $request );
+		$data[] = $this->prepare_response_for_collection( $item );
+
+		return rest_ensure_response( $data );
+	}
+
+	/**
+	 * Prepare a report sales object for serialization.
+	 *
+	 * @param null $_
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response $response Response data.
+	 */
+	public function prepare_item_for_response( $_, $request ) {
+		// Set date filtering.
+		$filter = array(
+			'period'   => $request['period'],
+			'date_min' => $request['date_min'],
+			'date_max' => $request['date_max'],
+		);
+		$this->setup_report( $filter );
+
+		// New customers.
+		$users_query = new WP_User_Query(
+			array(
+				'fields' => array( 'user_registered' ),
+				'role'   => 'customer',
+			)
+		);
+
+		$customers = $users_query->get_results();
+
+		foreach ( $customers as $key => $customer ) {
+			if ( strtotime( $customer->user_registered ) < $this->report->start_date || strtotime( $customer->user_registered ) > $this->report->end_date ) {
+				unset( $customers[ $key ] );
+			}
+		}
+
+		$total_customers = count( $customers );
+		$report_data     = $this->report->get_report_data();
+		$period_totals   = array();
+
+		// Setup period totals by ensuring each period in the interval has data.
+		for ( $i = 0; $i <= $this->report->chart_interval; $i ++ ) {
```

Comment:
I don't think the coding standards address it directly, but I think we leave spaces off for incrementing and decrementing (so `$i++` should be used here)